builds/custom-builds-buildah.adoc

```diff
@@ -0,0 +1,34 @@
+[id='custom-builds-buildah']
+= Running custom builds with Buildah
```

Comment:
I might change the title to discuss the Docker socket instead of Buildah. I think users might care more about getting the Docker socket than that they have to use Buildah to do it.
"Adding a Docker socket to custom builds with Buildah"
"Adding a Docker socket to custom builds"
"Running custom builds that need Docker sockets"